modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -82,6 +103,165 @@ export async function createCombinedKey(
+ */
+export async function createUserSignatureShare(
+  oShare: OShare,
+  dShare: DShare,
+  message: Buffer
+): Promise<SignatureShare> {
+  if (oShare.i === ShareKeyPosition.BITGO) {
+    throw new Error(`Invalid OShare, doesn't belong to the User`);
+  }
+
+  if (dShare.i === ShareKeyPosition.BITGO || dShare.j !== ShareKeyPosition.BITGO) {
+    throw new Error(`Invalid DShare, doesn't seem to be from BitGo`);
+  }
+  return MPC.sign(message, oShare, dShare);
+}
+
+/**
+ * Sends Share To Bitgo
+ * @param {BitGoBase} bitgo - the bitgo instance
+ * @param {String} walletId - the wallet id  *
+ * @param {String} txRequestId - the txRequest Id
+ * @param {SignatureShareRecord} signatureShare - a Signature Share
+ * @returns {Promise<SignatureShareRecord>} - a Signature Share
+ */
+export async function sendShareToBitgo(
+  bitgo: BitGoBase,
+  walletId: string,
+  txRequestId: string,
+  shareType: SendShareType,
+  share: SShare | MUShare | KShare,
+  dShare?: DShare
+): Promise<SendShareToBitgoRT> {
+  if (shareType !== SendShareType.SShare && share.i !== ShareKeyPosition.BITGO) {
+    throw new Error('Invalid Share, is not from User to Bitgo');
+  }
+  const signatureShare: SignatureShareRecord = {
+    from: SignatureShareType.USER,
+    to: SignatureShareType.BITGO,
+    share: '',
+  };
+
+  let responseFromBitgo: SendShareToBitgoRT;
+
+  switch (shareType) {
+    case SendShareType.KShare:
+      const kShare = share as KShare;
+      signatureShare.share = kShare.k + kShare.n;
+      const AShareRecord = await sendSignatureShare(bitgo, walletId, txRequestId, signatureShare);
+      if (AShareRecord.share.length < 4608) {
+        throw new Error('Invalid AShare from Bitgo');
+      }
+      responseFromBitgo = {
+        i: kShare.j,
+        j: 3,
+        k: AShareRecord.share.substring(0, 1536),
+        alpha: AShareRecord.share.substring(1536, 3072),
+        mu: AShareRecord.share.substring(3072, 4608),
+        n: AShareRecord.share.substring(4608),
+      };
+      break;
+    case SendShareType.MUShare:
+      const muShare = share as MUShare;
+      signatureShare.share = muShare.alpha + muShare.mu;
+      const dShareRecord = await sendSignatureShare(bitgo, walletId, txRequestId, signatureShare);
+      if (dShareRecord.share.length !== 130) {
+        throw new Error('Invalid DShare from Bitgo');
+      }
+      responseFromBitgo = {
+        i: muShare.j,
+        j: 3,
+        delta: dShareRecord.share.substring(0, 64),
+        Gamma: dShareRecord.share.substring(64, 130),
+      };
+      break;
+    case SendShareType.SShare:
+      if (!dShare) {
+        throw new Error('DShare not provided');
+      }
+      if (dShare.i !== ShareKeyPosition.BITGO || dShare.j === ShareKeyPosition.BITGO) {
+        throw new Error('Invalid Dshare provided');
+      }
+      const sShare = share as SShare;
+      signatureShare.share = sShare.r + sShare.s + sShare.y + dShare.delta + dShare.Gamma;
```

Comment:
Is this a valid way to send multiple shares? Not a blocker since this capability is not supported yet, but this PR (https://github.com/BitGo/bitgo-microservices/pull/20947) will allow us to pass in multiple signature share records once it's merged so maybe we can change this to use that format in the future

Reply:
Okay nice, But Let's have it tracked via another ticket and handle the new case once its merged.